.azure/pipelines/build.yaml

```diff
@@ -210,7 +237,7 @@ jobs:
         ArtifactName: nuget
 
 # Tests
-- ${{ if eq(parameters.skip_test, false) }}:
+- ${{ if and(eq(parameters.skip_test, false), ne(variables.runCodeQL3000, 'true')) }}:
```

Comment:
As-is, tests are disabled in scheduled runs even if the CodeQL3000 tasks do nothing (but are enabled). It may be possible to change this condition so that tests run when the CodeQL tasks no-op. Is that necessary❔

Reply:
I am not quite understanding the purpose of this, is it a security measure?
Under what circumstances does CodeQL3000 do nothing (runCodeQL3000 != true)?

Reply:
This is to ensure that builds running the CodeQL3000 tasks aren't shown as failures due to unrelated test issues i.e. to make it easy to see whether the CodeQL3000 reporting is working. I asked the question above because Orleans is unusual in having its regular pipeline already do scheduled builds, elsewhere I _added_ a schedule just for CodeQL3000.

The CodeQL3000 tasks are included in a build only when `runCodeQL3000` is `true`. But, as [commented on above](https://github.com/dotnet/orleans/pull/8159/files#diff-66fc07ffa2464dbb58010aa7c74405252e1802e0951c8f0240698804ff0af4b3R101), the tasks do nothing of interest when the analysis is considered up to date. In that case, the pipeline will do a regular build but not test it. I'm not positive I can wire this up differently and wanted to know how important testing is considered in official builds (considering the public scheduled builds run the same tests) before trying to do something different here.

Reply:
Should mention "tests are disabled in scheduled runs" applies only to _internal_ scheduled runs. Public scheduled runs will still execute tests unconditionally.